graphprot/NeuralNet.py

```diff
@@ -112,8 +114,29 @@ def __init__(self, database, Net,
             self.loss = MSELoss()
 
         elif self.task == 'class':
+            self.weights=None
+
+            # assign weights to each class in case of unbalanced dataset
+            # User weights definition
+            if type(self.class_weights) == list :
+                self.weights = self.class_weights
+
+            # Automatic definition base on the classes proportions in the input training set
+            elif self.class_weights == True :
+                targets_all = []
+                
+                for batch in self.train_loader:
+                    targets_all.append(batch.y)  
+                
+                targets_all = torch.cat(targets_all).squeeze().tolist()
+                self.weights = torch.tensor([targets_all.count(i) for i in self.classes], dtype=torch.float32)
+                print('class occurences: {}'.format(self.weights))
+                self.weights = 1.0 / self.weights
+                self.weights = self.weights / self.weights.sum()
+                print('class weights: {}'.format(self.weights))
+
```

Comment:
I would most likely encapsulate that in a small function `compute_class_weights` for clarity